servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -367,8 +368,16 @@ public boolean isDirectory(long id) {
    */
   public long getFileId(TachyonURI path) throws InvalidPathException {
     synchronized (mInodeTree) {
-      Inode inode = mInodeTree.getInodeByPath(path);
-      return inode.getId();
+      try {
+        Inode inode = mInodeTree.getInodeByPath(path);
+        return inode.getId();
+      } catch (InvalidPathException e) {
+        try {
+          return loadMetadata(path, true);
+        } catch (Exception e2) {
+          throw e;
```

Comment:
Also I dislike having to use exceptions to convey the fact the operation failed, since we are doing it speculatively (failure is an expected outcome), but this can be addressed later.